packages/elements-react/src/components/form/form.tsx

```diff
@@ -264,13 +304,42 @@ export function OryForm({ children }: OryFormProps) {
         })
         break
       }
-      case FlowType.Settings:
+      case FlowType.Settings: {
+        const submitData: UpdateSettingsFlowBody = {
+          ...(data as unknown as UpdateSettingsFlowBody),
+        }
+
+        if ("totp_unlink" in submitData) {
+          submitData.method = "totp"
+        }
+
+        if (
+          "lookup_secret_confirm" in submitData ||
+          "lookup_secret_reveal" in submitData ||
+          "lookup_secret_regenerate" in submitData
+        ) {
+          submitData.method = "lookup_secret"
+        }
+
+        if ("link" in submitData || "unlink" in submitData) {
+          submitData.method = "oidc"
+        }
+
+        if ("webauthn_remove" in submitData) {
+          submitData.method = "webauthn"
+        }
+
+        if ("passkey_remove" in submitData) {
+          submitData.method = "passkey"
+        }
```

Comment:
Do we actually need this? AFAIK we set the method correctly in Kratos (it's not always needed)

Reply:
Yes, we do, because the SDK doesn't know what to do if we don't set the method in the submitted data. 
<img width="670" alt="image" src="https://github.com/user-attachments/assets/f33f5f52-45b4-4e92-b798-87ad8a038a6c">

So Kratos might be able to deal with it, but the SDK isn't.